sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py

```diff
@@ -0,0 +1,1210 @@
+                key_usage = [k.value for k in self.key_properties.key_usage]
+            else:
+                key_usage = None
+            x509_certificate_properties = models.X509CertificateProperties(
+                subject=self.subject_name,
+                ekus=self.key_properties.ekus if self.key_properties else None,
+                subject_alternative_names=models.SubjectAlternativeNames(
+                    emails=self.san_emails,
+                    upns=self.san_upns,
+                    dns_names=self.san_dns_names
+                ),
+                key_usage=key_usage,
+                validity_in_months=self.validity_in_months
+            )
+        else:
+            x509_certificate_properties = None
+
+        if (self.key_properties and
+                (self.key_properties.exportable or
+                self.key_properties.key_type or
+                self.key_properties.key_size or
+                self.key_properties.reuse_key or
+                self.key_properties.curve)):
+            key_properties = models.KeyProperties(
+                exportable=self.key_properties.exportable,
+                key_type=self.key_properties.key_type,
+                key_size=self.key_properties.key_size,
+                reuse_key=self.key_properties.reuse_key,
+                curve=self.key_properties.curve
+            )
+        else:
+            key_properties = None
+
+        if self.content_type:
+            secret_properties = models.SecretProperties(content_type=self.content_type.value)
+        else:
+            secret_properties = None
+
+        policy_bundle = models.CertificatePolicy(
+            id=self.id,
+            key_properties=key_properties,
+            secret_properties=secret_properties,
+            x509_certificate_properties=x509_certificate_properties,
+            lifetime_actions=lifetime_actions,
+            issuer_parameters=issuer_parameters,
+            attributes=attributes
+        )
+        return policy_bundle
+
+    @classmethod
+    def _from_certificate_policy_bundle(cls, certificate_policy_bundle):
+        # type: (models.CertificatePolicy) -> CertificatePolicy
+        """Construct a CertificatePolicy from an autorest-generated CertificatePolicy"""
+        if certificate_policy_bundle.lifetime_actions:
+            lifetime_actions = [
+                LifetimeAction(
+                    action_type=item.action.action_type,
+                    lifetime_percentage=item.trigger.lifetime_percentage,
+                    days_before_expiry=item.trigger.days_before_expiry,
+                )
+                for item in certificate_policy_bundle.lifetime_actions
+            ]
+        else:
+            lifetime_actions = None
+        key_properties_bundle = certificate_policy_bundle.key_properties
+        # pylint:disable=too-many-boolean-expressions
+        if (key_properties_bundle and
+                (key_properties_bundle.exportable or
```

Comment:
`key_properties_bundle` is from a Key Vault API response. I doubt Key Vault would send `"key_properties": {}`, so I think you can delete this second condition.

Reply:
that meks sense, I've changed it and it's a lot cleaner now thanks!